src/test/java/io/jenkins/pluginhealth/scoring/probes/ProbeEngineTest.java

```diff
@@ -92,4 +95,40 @@ public void shouldNotSaveErrors() {
 
         verify(plugin, never()).addDetails(any());
     }
+
+    @Test
+    public void shouldBeAbleToGetPreviousContextResultInExecution() throws Exception {
+        final Plugin plugin = spy(Plugin.class);
+        final Probe probeOne = new Probe() {
+            @Override
+            protected ProbeResult doApply(Plugin plugin) {
+                return ProbeResult.success(key(), "This is ok");
+            }
+
+            @Override
+            protected String key() {
```

Comment:
Java Compiler is complaining about `ERROR: key() in cannot override key() in io.jenkins.pluginhealth.scoring.probes.Probe`. I'm guessing it's because you've used an Anonymous Subclass, am I right?

Reply:
no because it should be a `public` method now.